File: 0122-Best Time to Buy and Sell Stock II/c++/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit = 0;
        for(int idx = 1; idx < prices.size(); ++idx) 
            if(prices[idx] > prices[idx-1])
                max_profit += prices[idx] - prices[idx-1];
        return max_profit;
    }
};

// Runtime: 8 ms, faster than 96.60% of C++ online submissions for Best Time to Buy and Sell Stock II.
// Memory Usage: 13.2 MB, less than 41.65% of C++ online submissions for Best Time to Buy and Sell Stock II.

File: 0122-Best Time to Buy and Sell Stock II/golang/solution.go
package golang

func maxProfit(prices []int) int {
	max_profit := 0
	for idx := 1; idx < len(prices); idx++ {
		if prices[idx] > prices[idx-1] {
			max_profit += prices[idx] - prices[idx-1]
		}
	}
	return max_profit
}

// Runtime: 0 ms, faster than 100.00% of Go online submissions for Best Time to Buy and Sell Stock II.
// Memory Usage: 3.2 MB, less than 16.63% of Go online submissions for Best Time to Buy and Sell Stock II.
